src/buf.c

```diff
@@ -51,6 +51,7 @@ struct Buf defs_buf;		/**< for #define's autogenerated. List of strings. */
 struct Buf yydmap_buf;		/**< string buffer to hold yydmap elements */
 struct Buf m4defs_buf;          /**< m4 definitions. List of strings. */
 struct Buf top_buf;             /**< contains %top code. String buffer. */
+struct Buf alloc_buf;           /**< List of strings that need to be freed. */
```

Comment:
@whydoubt  
Could you please explain why is this change needed? If I read correctly, buf_append() **does not** allocate additional buffer, but *reallocates* existing buffer instead.

You know how realloc() works, right? When realloc() succeeds, the data in the old buffer will be copied to new one, and the old buffer get free'd.

So I don't see the purpose of this `alloc_buf`  tracking all the things. Besides, I think your implementation is incorrect... (see below)